columnflow/calibration/cms/jets.py

```diff
@@ -111,58 +115,130 @@ def ak_evaluate(evaluator: correctionlib.highlevel.Correction, *args) -> float:
         })
 
     :param external_files: Dictionary containing the information about the file location
     :return: path or url to correction file(s)
     """ # noqa
-    return external_files.jet_jerc
+
+    # get config
+    try_attrs = ("get_jec_config", "get_jer_config")
+    jerc_config = None
+    for try_attr in try_attrs:
+        try:
+            jerc_config = getattr(self, try_attr)()
+        except AttributeError:
+            continue
+        else:
+            break
+
+    # fail if not found
+    if jerc_config is None:
+        raise ValueError(
+            "could not retrieve jer/jec config, none of the following methods "
+            f"were found: {try_attrs}",
+        )
+
+    # first check config for user-supplied `external_file_key`
+    ext_file_key = jerc_config.get("external_file_key", None)
+    if ext_file_key is not None:
+        return external_files[ext_file_key]
+
+    # if not found, try to resolve from jet collection name and fail if not standard NanoAOD
+    if self.jet_name not in get_jerc_file_default.map_jet_name_file_key:
+        available_keys = ", ".join(sorted(get_jerc_file_default.map_jet_name_file_key))
+        raise ValueError(
+            f"could not determine external file key for jet collection '{self.jet_name}', "
+            f"name is not one of standard NanoAOD jet collections: {available_keys}",
+        )
+
+    # return external file
+    ext_file_key = get_jerc_file_default.map_jet_name_file_key[self.jet_name]
+    return external_files[ext_file_key]
+
+
+# default external file keys for known jet collections
+get_jerc_file_default.map_jet_name_file_key = {
+    "Jet": "jet_jerc",
+    "FatJet": "fatJet_jerc",
+}
 
 
 def get_jec_config_default(self) -> DotDict:
-    """Load config relevant to the JEC corrections.
+    """
+    Load config relevant to the jet energy corrections (JEC).
 
-    By default, this is extracted from the current *config_inst*:
+    By default, this is extracted from the current *config_inst*,
+    assuming the JEC configurations are stored under the 'jec'
+    aux key. Separate configurations should be specified for each
+    jet collection, using the collection name as a key. For example,
+    the configuration for the default jet collection ``Jet`` will
+    be retrieved from the following config entry:
 
     .. code-block:: python
 
-        self.config_inst.x.jec
+        self.config_inst.x.jec.Jet
 
     Used in :py:meth:`~.jec.setup_func`.
 
-    :return: Dictionary containing configurations for JEC callibrations
+    :return: Dictionary containing configuration for jet energy calibration
     """
-    return self.config_inst.x.jec
+    jec_cfg = self.config_inst.x.jec
+
+    # check for old-style config
+    if self.jet_name not in jec_cfg:
+        # if jet collection is `Jet`, issue deprecation warning
+        if self.jet_name == "Jet":
+            logger.warning_once(
```

Comment:
`warning_once` accepts two arguments with the first one being an identifier to check if this type of message should be skipped when logged again. E.g. `f"{id(self)}_depr_jec_config"`.

Reply:
Added as suggested.